pkg/utils/chart.go

```diff
@@ -328,3 +344,23 @@ func getReleaseName(hr *appsapi.HelmRelease) string {
 	}
 	return releaseName
 }
+
+func replaceCRD(cfg *action.Configuration, targetChart *chart.Chart) error {
+	queue := []*chart.Chart{targetChart}
+	for _, c := range queue {
```

Comment:
It is not memory safe to mutate `queue` in a for loop, especially when we've got a chart with multiple dependencies and crds.

I'd suggest using recursion algorithm to handle this.